test/replication-luatest/gh_9399_raft_leader_steps_off_on_ER_WAL_IO_test.lua

```diff
@@ -0,0 +1,92 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group('raft-leader-steps-off-on-ER_WAL_IO')
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = 10}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+g.before_each(function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.cluster = cluster:new({})
+
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('master', cg.cluster.id),
+            server.build_listen_uri('replica', cg.cluster.id),
+        },
+        election_mode = 'candidate',
+        replication_synchro_quorum = 1,
+        replication_synchro_timeout = 100000,
+        replication_timeout = 0.1,
+        election_fencing_mode = 'off',
+    }
+    cg.master = cg.cluster:build_and_add_server({
+        alias = 'master',
+        box_cfg = box_cfg
+    })
+    box_cfg.election_mode = 'off'
+    cg.replica = cg.cluster:build_and_add_server({
+        alias = 'replica',
+        box_cfg = box_cfg
+    })
+    cg.cluster:start()
+    cg.master:wait_until_election_leader_found()
+    cg.replica:wait_until_election_leader_found()
+
+    cg.master:exec(function()
+        box.schema.space.create('test', {is_sync = true})
+        box.space.test:create_index('pk')
+    end)
+    wait_pair_sync(cg.replica, cg.master)
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_cluster_elect_new_leader = function(cg)
+    cg.replica:exec(function()
+        box.cfg{election_mode='candidate'}
+    end)
+    cg.master:exec(function()
+        t.assert_equals(box.info.election.state, 'leader')
+        box.error.injection.set('ERRINJ_WAL_IO', true)
```

Comment:
The errinj must be unset back to false after the test case. Otherwise if any other test case would run after this one on the same instance, it would fail. In this case right now it is not so, but better not leave any errinj dangling anywhere. Code sometimes gets moved or copy-pasted and the bug could actually migrate to a place where it would manifest.

Reply:
The master falls in this test, it is called panic. Therefore, there is no way to unset it back to false manually. 
I can do `box.error.injection.set('ERRINJ_WAL_IO_COUNTDOWN', 1)` and then `ERRINJ_WAL_IO` will automatically be set to false before the instance falls. But is this necessary if the instance cannot be reused anyway?

Reply:
So the instance crashes and the test passes? That doesn't sound right in terms of how luatest is handling it. Who knows where else our instances are crashing in the tests and we don't see it. Please, file a ticket in luatest repo if it isn't already there.

Reply:
I asked the developers of luatest. They said it was normal. luatest only detects segfaults, it ignores panics (non-zero return code). [luatest doesn't detect crash at exit](https://github.com/tarantool/luatest/issues/252) This is done so that you can test the panic call on an instance, as I do in this test.